app/main.py

```diff
@@ -1,2 +1,159 @@
+from fractions import Fraction
+from typing import Any, Hashable, Iterable, Optional
+
+
 class Dictionary:
-    pass
+    class __Node:
+        def __init__(
+                self,
+                key: Optional[Hashable] = None,
+                value: Optional[Any] = None,
+                key_hash: Optional[int] = None
+        ) -> None:
+            self.key = key
+            self.value = value
+            self.key_hash = key_hash
+            self.deleted = False
+
+    def __init__(
+            self,
+            initial_data: Optional[Iterable[tuple[Hashable, Any]]] = None,
+            **kwargs
+    ) -> None:
+        self.__capacity = 8
+        self.__size = 0
+        self.__threshold = int(Fraction(2, 3) * self.__capacity)
+        self.__table = [Dictionary.__Node()] * self.__capacity
```

Comment:
would be more optimal to fill out empty hash table with None values instead of creating `capacity` times instances of Node

Reply:
Yeah I see. You can avoid all this if you make changes to your calculate index function. Here how this could look like.
Imagine that your hash table is filled out with None values.
```
If you look to set a Node:  if table[index] is None -> free cell - you can use it to set a new NOde. If not check cell index+1 -> repeat until you find one you need
If you looking for a value by index: if table[index] is occupied -> check if node that is there has the same key and hash that you looking for and return that index if matching. If not check cell index+1 -> repeat until you find one you need

Reply:
You basically have the same logic in your calculate index method, just need to change  a little bit to make it work

Reply:
@Oleksl888 
How then can I implement the __del__ method rightly? And deal with the getting colissioned values?
I understand what you want me to do, but this doesn't solve the issue.

If I have Nones in my hash table instead of Nodes then, for example:
1) if I ad an element: my_dict[0] = "a" --> this will assign to my hash_table[0] a Node object.
2) Then I want to add one more element: my_dict[8] = "b" --> a colission will occure. Because it will want to assign to my hash_table[0] a new Node. But in my calculate function I look to the next index. And find that it is None and assign to it: hash_table[1] a Node object.
3) In this case I have no problem to access my_dict[8], because I look to my_dict[0], I see it is not empty and keys are different, then I look to my_dict[1].

But here is the problem:
4) I delete my_dict[0] (it was with value "a"). I set None to hash_table[0]
5) Then if I want to access my_dict[8] -- I have no access. Because I look to hash_table[0] and see that it is None. I return KeyError
But actually there is this key in my dictionary...
What if I add my_dict[16], my_dict[24] and so on? 

I need somehow then to flag my deleted indexes. If the index was deleted and before colission happened -- I can move further and check other indexes in order to get the value.

Reply:
Ok you can keep it as is because your implementation actually works. However it is not most efficient one - you fill out your hash table with lots of garbage values which takes extra memory.

Reply:
I assure you that it is possible to do without marking deleted indexes - basically the fact that a table has None at that index you pointing to means that its either free memory you can use in case if you want to set value or if it's not None you check if key and hash value is correct. Then if something doesn't add up you move to one cell on the right..

Reply:
Thank you!